src/analyser.cpp

```diff
@@ -2524,9 +2528,29 @@ void Analyser::AnalyserImpl::analyseModel(const ModelPtr &model)
                             || relevantCheck;
         }
 
-        if (!relevantCheck && !checkNlaSystems) {
+        if (((loopNb == 1) || (loopNb == 3)) && !relevantCheck) {
+            ++loopNb;
+
             relevantCheck = true;
             checkNlaSystems = true;
+        } else if ((loopNb == 2) && !relevantCheck) {
+            // We have gone through the two loops and we still have some unknown
+            // variables, so we consider them as initialised and we go through
```

Comment:
should this mention external variables? i.e., `so we consider external variables as initialised and we go...`

Reply:
No, but I agree that this could be made clearer, hence commit 31f3d228 (which also improves the analyser a bit).